src/partition/ReceivedPartition.hpp

```diff
@@ -104,8 +99,28 @@ class ReceivedPartition : public Partition {
   /// Min global vertex IDs of remote connected ranks
   std::vector<int> _remoteMinGlobalVertexIDs;
 
-  template <typename T>
-  friend void PartitionTests::ReceivedPartitionTests::testParallelSetOwnerInformation(T &mesh, int dimensions);
+  // Make the fixture friend of this class
+  friend struct receivedPartitionFixture;
+};
+
+/*
+ * @brief A fixture that is used to access private functions of the receivedPartition class.
+ *
+ * The fixture can be used to call private functions for individual testing. 
+ */
+struct receivedPartitionFixture {
+  void createOwnerInformation(ReceivedPartition &part)
```

Comment:
Making these functions static will allow to call the functions without creating an instance of the fixture first.

```suggestion
  static void createOwnerInformation(ReceivedPartition &part)
```

Reply:
when defined static, it is not able to call class members and it throws error.

Reply:
done